repository/feature_flag_repo.go

```diff
@@ -25,11 +25,17 @@ func NewFeatureFlagRepo(c cache.Cache, config map[domain.FeatureFlagKey][]domain
 		return fcr, nil
 	}
 
-	for key, cfg := range config {
-		ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	for key, flags := range config {
+		// Timeout should be set based on the number of flags in the environment rather than
+		// a fixed timeout for each environment. This means that environments with more flags
+		// get a longer timeout, if we just have a fixed timeout for each environment e.g. 10 seconds
+		// then that might not be long enough for environments with a large number of flags.
+		timeout := 10 * len(config)
```

Comment:
Is this 10ms if we have one flag? Maybe we should set a minimum timeout e.g. the old 10 seconds but if we're above some data size threshold then let it be greater. If we have redis connected over a network somewhere we might timeout before we can even speak to it for smaller data sizes.

Reply:
How long are you seeing based on our prod data? 10s per flag seems a long time I wonder if in some error scenario it would leave the proxy hanging forever on startup e.g. with 200 flags that gives a timeout of over 30 minutes

Reply:
For targets it seems to get about 250 of them in for our UAT environment before it times out. Not sure yet how many flags it would get in before it times out because we exit out of main once we fail to load the targets in.